src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/Bedrock/InvokeModelAsyncWrapper.cs

```diff
@@ -60,25 +61,31 @@ void InvokeTryProcessResponse(Task responseTask)
                 if (responseTask.IsFaulted)
                 {
                     HandleError(segment, invokeModelRequest, responseTask, agent);
+                    return;
```

Comment:
This seems to return before ending the segment.

Reply:
I originally ended the segment at the top of this method, but that seemed to result in no errors getting noticed. So I guess I should end it here just before returning?

Reply:
Yes, I think we should end it here before returning. I don't think there's a cleaner way.